File: app/src/main/java/com/unip/aps/aps_noticias/ui/activity/CriarContaActivity.java
package com.unip.aps.aps_noticias.ui.activity;

import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.unip.aps.aps_noticias.R;
import com.unip.aps.aps_noticias.app.ApsNoticiasApp;
import com.unip.aps.aps_noticias.model.UsuarioModel;
import com.unip.aps.aps_noticias.services.UsuarioService;

public class CriarContaActivity extends BaseActivity {

    public static final int REQUEST_CRIAR_CONTA = 124;

    private EditText ed_nome;
    private EditText ed_email;
    private EditText ed_password;
    private Button bt_criar_conta;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_criar_conta);

        ed_nome = (EditText) findViewById(R.id.ed_nome_criar_activity);
        ed_email = (EditText) findViewById(R.id.ed_email_criar_activity);
        ed_password = (EditText) findViewById(R.id.ed_senha_criar_activity);

        bt_criar_conta = (Button) findViewById(R.id.bt_conta_criar_activity);
        bt_criar_conta.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                clickCriarConta();
            }
        });
    }

    private void clickCriarConta(){
        String nome = ed_nome.getText().toString();
        String email = ed_email.getText().toString();
        String senha = ed_password.getText().toString();

        if (!TextUtils.isEmpty(nome) && !TextUtils.isEmpty(email) && !TextUtils.isEmpty(senha)) {

            UsuarioModel user = new UsuarioModel();
            user.setNomePerfil(nome);
            user.setEmail(email);
            user.setSenha(senha);

            criarConta(user);

        }else{
            Toast.makeText(this, R.string.preencher_campos, Toast.LENGTH_SHORT).show();
        }
    }

    private void criarConta(UsuarioModel user){
        ApsNoticiasApp.getInstance().saveUser(user);
        setResult(RESULT_OK);
        finish();

//        showProgressDialog(true, getString(R.string.criando_usuario));
//        UsuarioService.criarUsuario(usuario, new UsuarioService.OnCriarUsuario() {
//            @Override
//            public void onSuccess() {
//                showProgressDialog(false, null);
//
//                ApsNoticiasApp.getInstance().saveUser(user);
//                setResult(RESULT_OK);
//                finish();
//            }
//
//            @Override
//            public void onError(String error) {
//                showProgressDialog(false, null);
//                Toast.makeText(CriarContaActivity.this, error, Toast.LENGTH_SHORT).show();
//            }
//        });
    }
}

File: app/src/main/java/com/noticias_now/ui/activity/CriarContaActivity.kt
package com.noticias_now.ui.activity

import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.EditText
import br.com.ricarlo.common.inapp.review.ReviewFlow
import br.com.ricarlo.common.inapp.review.ReviewViewModel
import br.com.ricarlo.common.util.extensions.openPlayStore
import com.google.android.play.core.review.ReviewManager
import com.noticias_now.R
import com.noticias_now.app.ApsNoticiasApp
import com.noticias_now.model.UsuarioModel
import com.noticias_now.services.UsuarioService
import com.noticias_now.services.UsuarioService.OnCriarUsuario
import com.noticias_now.util.showToast
import org.koin.android.ext.android.inject

class CriarContaActivity : BaseActivity() {

    private var ed_nome: EditText? = null
    private var ed_email: EditText? = null
    private var ed_password: EditText? = null
    private var bt_criar_conta: Button? = null

    private val reviewManager: ReviewManager by inject()
    private val reviewViewModel: ReviewViewModel by inject()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_criar_conta)

        ed_nome = findViewById(R.id.ed_nome_criar_activity)
        ed_email = findViewById(R.id.ed_email_criar_activity)
        ed_password = findViewById(R.id.ed_senha_criar_activity)
        bt_criar_conta = findViewById(R.id.bt_conta_criar_activity)
        bt_criar_conta?.setOnClickListener { clickCriarConta() }


        reviewViewModel.reviewFlow.observe(this, { reviewFlow ->
            when (reviewFlow) {
                is ReviewFlow.LaunchInApp -> {
                    reviewManager.launchReviewFlow(this, reviewFlow.reviewInfo)
                            .addOnCompleteListener {
                                Log.e("launchReviewFlow", "$it")
                                if (!it.isSuccessful) {
                                    showToast(it.exception.message.orEmpty())
                                }
                            }
                }
                is ReviewFlow.LaunchOutApp -> {
                    openPlayStore()
                }
                is ReviewFlow.Error -> {
                    showToast(reviewFlow.exception.message.orEmpty())
                }
            }
        })
    }

    override fun onResume() {
        super.onResume()
        reviewViewModel.requestReviewFlow()
    }

    private fun clickCriarConta() {
        val nome = ed_nome?.text.toString()
        val email = ed_email?.text.toString()
        val senha = ed_password?.text.toString()
        if (nome.isNotEmpty() && email.isNotEmpty() && senha.isNotEmpty()) {
            val user = UsuarioModel()
            user.name = nome
            user.email = email
            user.senha = senha
            user.photo = ""
            criarConta(user)
        } else {
            showToast(getString(R.string.preencher_campos))
        }
    }

    private fun criarConta(user: UsuarioModel) {
        showProgressDialog(true, getString(R.string.criando_usuario))
        UsuarioService.criarUsuario(user, object : OnCriarUsuario {
            override fun onSuccess(usuario: UsuarioModel) {
                showProgressDialog(false, null)
                ApsNoticiasApp.instance?.saveUser(usuario)
                setResult(RESULT_OK)
                finish()
            }

            override fun onError(error: String) {
                showProgressDialog(false, null)
                showToast(error)
            }
        })
    }

    companion object {
        const val REQUEST_CRIAR_CONTA = 124
    }

}